scripts/generate_serialization.py

```diff
@@ -586,7 +597,12 @@ def generate_class_code(class_entry):
             deserialize_template_str = deserialize_element_class_base.replace(
                 '${BASE_PROPERTY}', entry.base.replace('*', '')
             ).replace('${DERIVED_PROPERTY}', entry.type.replace('*', ''))
-        type_name = replace_pointer(entry.type)
+        if entry.custom_serialize_property == True and is_pointer(entry.type):
```

Comment:
I don't quite follow the need for this extra code here - why turn entry types that are not marked as an optional ptr into an optional ptr here?

Reply:
Perhaps it's necessary to provide further context:
<https://github.com/Tishj/duckdb/pull/129>
This is the PR that I need this for, I figured this was stand alone and perhaps controversial (my hunch was right) enough to be a separate PR

Reply:
So I talked about it a bit with Max, Carlo and Sam and they proposed a good idea: `optionally_owned_ptr`
That special cases this relationship in a way that we probably don't have to touch other serialization code to support it

Basically it encapsulates these two variables we have here:
`owned`
`ref`

If it's constructed with a unique_ptr, it sets owned and ref, if it's constructed with a reference, it only sets ref